pic-sure-util/src/main/java/edu/harvard/dbmi/avillach/util/Utilities.java

```diff
@@ -25,8 +27,19 @@ public static HttpClientContext buildHttpClientContext() {
     }
 
     public static String getRequestSourceFromHeader(HttpHeaders headers) {
-        if (headers == null) return "";
-        return Optional.ofNullable(headers.getHeaderString("requestSource")).orElse("");
+        if (headers == null) return "headers are null";
+        return Optional.ofNullable(headers.getHeaderString("request-source")).orElse("request-source header is null");
```

Comment:
This is an interesting use of Optional. Usually you see it as a means to both mark return types as nullable and force api consumers to handle nulls. Using an optional purely as a means of control flow is a bit different, but I don't see a reason not to do it. 

You could go a little bit further with this if you're OK with a more general error message:

```suggestion
        return Optional.ofNullable(headers)
            .map(h ->  h.getHeaderString("request-source"))
            .orElse("request-source header is null or headers object is null");
```

Reply:
I prefer the less general error message.  I don't want to have to debug further to determine if it was the header object or the request-source. Additionally, I will convert this from a Optional to a ternary.